### ISSUE TITLE
[Feature Request] Specify TLS min version through configuration

### ISSUE THREAD

**Issue by kklemmicknvidia** [docker/docker#30258]
Pull request 8588 set the minimum TLS version to 1.0 in code: https://github.com/docker/docker/pull/8588

However, our security department wants to eliminate the use of TLSv1 entirely and docker has been identified as having this security risk. Is it possible to set the TLS version to 1.1 through the config without recompiling docker?

Ideally this would be something in the daemon.json like:
        "tlsminversion": tls.VersionTLS11

But it doesn't look like that's possible at the moment.

Any suggestions? Would this be a reasonable feature request?

Thanks,
Kevin

**Comment by justincormack** [comment]
This feature does not exist at present. It should be fairly easy to set those from a config value rather than a constant.

cc @cyli @riyazdf

**Comment by cyli** [comment]
Agreed, although we should probably ensure that whatever value is passed is >= TLSv1.

**Comment by kklemmicknvidia** [comment]
Thanks guys, edited this to include "Feature Request" tag.

**Comment by adshmh** [comment]
#dibs

**Comment by adshmh** [comment]
The PR against go-connections is needed so the required minimum TLS version can be passed as an option to tlsconfig. If that PR is approved, I will submit a PR on docker/docker for handling the configuration parameter/flag. 

On a side note, I was wondering (as a separate PR), if the CommonTLSOptions struct used by the daemon should also be used by the client for consistency and to have one less conversion. This would also reduce the dependency on tlsconfig from both client and the daemon to only the daemon.

**Comment by tiborvass** [comment]
Oh boy...

**Comment by adshmh** [comment]
As @tiborvass pointed out, I put the expression 'needed to fix docker/docker#30258' in the git commit message, which will cause this issue to get closed every time the commit gets pushed by a maintainer to their fork of go-connections.

I am still working to submit 2 more PRs that are needed for this issue: one to vendor the go-connections page and one for the changes needed on docker side.

**Comment by thaJeztah** [comment]
Yes, GitHub can be quite, erm, "pro-active" there. I usually put the `fix xx` only in the PR description, and not in the actual commit message for that reason :smile:

No worries though, we'll just have to watch this issue and reopen if it's automatically closed prematurely
